@here/harp-materials/lib/WebGL1RawShaderMaterial.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as THREE from "three";
+
+function convertVertexShaderToWebGL2(vertexShader: string): string {
+    return (
+        ["#define attribute in", "#define varying out", "#define texture2D texture"].join("\n") +
+        "\n" +
+        vertexShader
+    );
+}
+
+function convertFragmentShaderToWebGL2(fragmentShader: string): string {
+    return (
+        [
+            "#define varying in",
+            "out highp vec4 pc_fragColor;",
+            "#define gl_FragColor pc_fragColor",
+            "#define gl_FragDepthEXT gl_FragDepth",
+            "#define texture2D texture",
+            "#define textureCube texture",
+            "#define texture2DProj textureProj",
+            "#define texture2DLodEXT textureLod",
+            "#define texture2DProjLodEXT textureProjLod",
+            "#define textureCubeLodEXT textureLod",
+            "#define texture2DGradEXT textureGrad",
+            "#define texture2DProjGradEXT textureProjGrad",
+            "#define textureCubeGradEXT textureGrad"
+        ].join("\n") +
+        "\n" +
+        fragmentShader
+    );
+}
+
+/**
+ * Material that converts webgl1 shaders to webgl2 compatible code when renderer uses a WebGL2
+ * context.
+ */
+export class WebGL1RawShaderMaterial extends THREE.RawShaderMaterial {
```

Comment:
The name is a bit confusing IMO, it seems to imply that the shader only works with WebGL1, however based on the comment, it seems to do the opposite.

Reply:
This is the base class of all the materials using webgl1 compatible shaders, hence the name.
It does convert the shaders to webgl2 if necessary though. Please suggest an alternative name.

Reply:
I think just "RawShaderMaterial" is fine, the conversion of the shader is just some technical detail that isn't really relevant to the user.

Reply:
ok.